packages/bpk-component-modal/src/BpkModalDialog.js

```diff
@@ -40,7 +40,7 @@ export type Props = {
   fullScreen: boolean,
   padded: boolean,
   dialogRef: (ref: ?HTMLElement) => void,
-  onClose: ?(event: SyntheticEvent<>) => void,
+  onClose: (event: SyntheticEvent<>) => void,
```

Comment:
Changed this due to the following flow error with mixing between components

<img width="979" alt="Screenshot 2020-08-04 at 14 50 36" src="https://user-images.githubusercontent.com/8831547/89301680-dca20980-d661-11ea-9cf7-62e8af975de3.png">

Changed the default to be a function to ensure that null is not passed and caused `BpkButtonLink` not to be used incorrectly